env/env_test.cc

```diff
@@ -1522,6 +1522,61 @@ TEST_F(EnvPosixTest, MultiReadNonAlignedLargeNum) {
   }
 }
 
+TEST_F(EnvPosixTest, MultiReadDirectIONonAlignedLargeNum) {
+  EnvOptions soptions;
+  soptions.use_direct_reads = true;
+  soptions.use_direct_writes = false;
+  std::string fname = test::PerThreadDBPath(env_, "testfile");
+
+  Random rnd(301);
+  std::unique_ptr<WritableFile> wfile;
+  size_t alignment = 0;
+  // Create file.
+  {
+    ASSERT_OK(env_->NewWritableFile(fname, &wfile, soptions));
+    auto data_ptr = NewAligned(4095, 'b');
+    Slice data_b(data_ptr.get(), 4095);
+    ASSERT_OK(wfile->PositionedAppend(data_b, 0U));
+    ASSERT_OK(wfile->Close());
+  }
+
+  for (uint32_t attempt = 0; attempt < 5; attempt++) {
+    const int num_reads = 1;
+    // Create requests
+    std::vector<std::string> scratches;
+    scratches.reserve(num_reads);
+    std::vector<ReadRequest> reqs(num_reads);
+
+    std::unique_ptr<RandomAccessFile> file;
+    ASSERT_OK(env_->NewRandomAccessFile(fname, &file, soptions));
+    alignment = file->GetRequiredBufferAlignment();
+    ASSERT_EQ(num_reads, reqs.size());
+
+    std::vector<std::unique_ptr<char, Deleter>> data;
+
+    size_t i = 0;
+    // Do alignment
+    reqs[i].offset = static_cast<uint64_t>(
+        TruncateToPageBoundary(alignment, static_cast<size_t>(0)));
+    reqs[i].len =
+        Roundup(static_cast<size_t>(0) + 4096, alignment) - reqs[i].offset;
```

Comment:
Is it always 4096?

Reply:
yes, I was requesting for only one read.